zebra/zebra_nhg.c

```diff
@@ -1959,6 +1960,52 @@ static int resolve_backup_nexthops(const struct nexthop *nexthop,
 	return 0;
 }
 
+/*
+ * This code is depending on the fact that a nexthop->ifindex is 0
```

Comment:
maybe before this caveat, it would be helpful to state what special case this is looking at: multiple connected routes for 'rn', where one of the other valid routes is a better match than 'match'.

Reply:
done